libraries/Filter/HarmonicNotchFilter.cpp

```diff
@@ -164,7 +182,7 @@ T HarmonicNotchFilter<T>::apply(const T &sample)
     }
 
     T output = sample;
-    for (uint8_t i = 0; i < _num_filters; i++) {
+    for (uint8_t i = 0; i < _num_enabled_filters; i++) {
```

Comment:
I wonder if it is possible that the _num_enabled_filters is perhaps 2 but it's actually the 1st and 3rd filters that are enabled.  I.e. are we sure that it will always be the final filters that are not enabled rather than a filter in the middle?

Reply:
@rmackay9 we only disable filters if they have a cutoff frequency above the nyquist limit. We only _create_ filters if they are in the harmonics bitmask - so in your example there would only be 2 filters created and whether they were enabled or not would depend on the center frequency. The center frequency of each filter is always higher than the previous so we could never enable the 3rd but have the 1st disabled.